dbms/src/Storages/Page/V3/BlobStore.cpp

```diff
@@ -115,6 +116,38 @@ void BlobStore::registerPaths()
     }
 }
 
+FileUsageStatistics BlobStore::getFileUsageStatistics() const
+{
+    FileUsageStatistics usage;
+
+    // Get a copy of stats map to avoid the big lock on stats map
+    const auto stats_list = blob_stats.getStats();
+
+    for (const auto & [path, stats] : stats_list)
+    {
+        (void)path;
+        for (const auto & stat : stats)
+        {
+            // We can access to these type without any locking.
+            if (stat->isReadOnly() || stat->isBigBlob())
+            {
+                usage.total_disk_size += stat->sm_total_size;
+                usage.total_valid_size += stat->sm_valid_size;
+            }
+            else
+            {
+                // Else the stat may being updated, acquire a lock to avoid data race.
+                auto lock = stat->lock();
```

Comment:
Can this lock be removed? we mat not need an exact value.

Reply:
1. Running under TSan may complain about that
2. We want a consistent value that total_size > valid size. Without lock, we may meet total_size < valid size, which is confusing
3. The frequency of calling getFileUsageStatistics is very low, call it only once every 30s 

So I would rather keep the lock here

Reply:
okay